server/src/main/java/org/elasticsearch/cluster/metadata/ClusterNameExpressionResolver.java

```diff
@@ -45,17 +45,26 @@ public ClusterNameExpressionResolver(Settings settings) {
      * Resolves the provided cluster expression to matching cluster names. This method only
      * supports exact or wildcard matches.
      *
-     * @param remoteClusters    the aliases for remote clusters
-     * @param clusterExpression the expressions that can be resolved to cluster names.
+     * @param remoteClusters            the aliases for remote clusters
+     * @param clusterExpression         the expressions that can be resolved to cluster names.
+     * @param expandExpressions         Whether expressions with wildcards are expanded
+     * @param ignoreNoneExistingAliases Whether none existing aliases are ignored
      * @return the resolved cluster aliases.
      */
-    public List<String> resolveClusterNames(Set<String> remoteClusters, String clusterExpression) {
+    public List<String> resolveClusterNames(Set<String> remoteClusters,
```

Comment:
> Maybe we could have two different methods to call? The strict one could return a String rather than a List. Maybe calling them resolveClusterNames (or rename it to resolveClusterNameExpression) and resolveClusterName would already explain the difference between the two?

:+1: But should we have a strict method for `RemoteClusterAware#groupClusterIndices(...)` too? Or just add a strict parameter to that method?

Reply:
good point, given that CCR always calls this method with a single item array, I would look into adding a specialized method for CCR. the following `if (clusters.isEmpty() == false) {` will also not be needed once the resolver is strict, and only one cluster can be returned (string rather than list). There may be a little code duplication introduced but it makes the API clearer, what do you think?

Reply:
This makes sense to me. I will be making these changes.